hera_cal/reflections.py

```diff
@@ -267,25 +239,445 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
                 freq_array = cal.freq_array
             else:
                 freq_array = self.freqs
             kwargs = dict([(k, getattr(cal, k)) for k in ['gain_convention', 'x_orientation',
                                                           'telescope_name', 'cal_style']])
+            add_to_history += "\nMerged-in calibration {}".format(input_calfits)
         else:
-            time_array = self.reflection_times
+            time_array = self.times
             freq_array = self.freqs
             kwargs = {}
 
-        uvc = io.write_cal(output_calfits, rgains, freq_array, time_array, flags=flags, quality=quals, 
-                           total_qual=tquals, outdir=os.path.dirname(output_calfits), zero_check=False, 
-                           overwrite=overwrite, history=version.history_string(add_to_history), **kwargs)
+        echo("...writing {}".format(output_calfits), verbose=verbose)
+        uvc = io.write_cal(output_calfits, rgains, freq_array, time_array, flags=flags,
+                           quality=quals, total_qual=tquals, zero_check=False,
+                           overwrite=overwrite, history=version.history_string(add_to_history),
+                           **kwargs)
+
+        if write_npz:
+            output_npz = os.path.splitext(output_calfits)[0] + '.npz'
+            if not os.path.exists(output_npz) or overwrite:
+                echo("...writing {}".format(output_npz), verbose=verbose)
+                np.savez(output_npz, delay=self.ref_dly, phase=self.ref_phs, amp=self.ref_amp,
+                         significance=self.ref_significance, times=self.times, freqs=self.freqs,
+                         lsts=self.lsts, antpos=self.antpos,
+                         history=version.history_string(add_to_history))
+
         return uvc
 
+    def pca_decomp(self, dly_range, dfft=None, flags=None, side='both', keys=None, overwrite=False, verbose=True):
+        """
+        Create a PCA-based model of the FFT data in dfft.
+
+        This is done via Singluar Value Decomposition on the input delay waterfall data
+        and results in self.umodes, self.vmodes, self.svals, and self.uflags.
+        The input dly_range forms the bounds of a tophat windowing applied to dfft
+        before taking the SVD, to narrow the information-content of the PCs to certain delays.
+
+        Args:
+            dly_range : len-2 tuple of positive delays in nanosec
+            dfft : DataContainer, holding delay-transformed data. Default is self.dfft
+            flags : DataContainer, holding dfft flags (e.g. skip_wgts). Default is None.
+            side : str, options=['pos', 'neg', 'both']
+                Specifies dly_range as positive delays, negative delays or both.
+            keys : list of tuples
+                List of datacontainer baseline-pol tuples to create model for.
+            overwrite : bool
+                If dfft exists, overwrite its values.
+        """
```

Comment:
There should be documentation of what objects are produced by this function and attached self, their types and shapes, etc.